### ISSUE TITLE
[FR] Supported skin flow & Supported skin speed

### ISSUE THREAD

**Issue by kulfuerst** [smartavionics/Cura#24]
Hey smartavionics,

it would be brilliant if it was possible to not only select the supported skin fan speed, but also flow and feedrate. Enabling supported skin bridge settings so to speak. I believe this may enable much lower support Z offset and therefore better dimensional accuracy of prints in the Z direction! For prints with larger nozzles it would also make the removal of supports easier which normally gets really difficult. It may also possibly (crossing fingers!) even enable a support Z offset of 0 when printing with a low flowrate while using large nozzles.
Long story short, this could potentially help improve supports a great amount. Thanks a ton for your awesome build :)

Best wishes

**Comment by smartavionics** [comment]
I have implemented this and it will appear in the next build. What platform are you using?

**Comment by kulfuerst** [comment]
Lightning fast as always. Im running your most recent build on Windows 10. Currently running all kinds of support tests so I really can't wait to check out what additional options this brings to the table :)
Thanks a lot

**Comment by smartavionics** [comment]
> Lightning fast as always.

Not really always true but often.

I will do a Windows build so you can try the new settings out, will be available later today.

**Comment by smartavionics** [comment]
New Windows build (1002) now on Dropbox.

**Comment by kulfuerst** [comment]
Okay, I've run some tests and they are extremely encouraging but also yield one problem that may be difficult to solve.
The images you can see below are the underside of the test print printed with a standard 0.4 mm Nozzle and a Support Z Distance of 0 mm (!!!) after I have removed the supports. The supports were very easy to remove and the underside of the print is as smooth as can be. As you can also see in the screenshot, the Supported Skin Settings only apply to supported skin, not supported walls, and this is causing the issue that supports that are directly under a skin region are easily removable even at a Support Z distanze of 0 mm, but the corresponding wall lines cannot be removed.
I wouldn't be surprised if this is very difficult to fix in the slicer and can understand if this can't be fixed without an unreasonable amount of work. If however, you can think of a fix to this issue that can apply the Supported Skin Settings also to supported walls, I believe this could result in a method of printing supports that are easily removable at a Z distance of 0 mm, yielding better looking undersides as well as more dimensionally accurate prints. My test results are promising.
Thank you for your help and I hope you find this equally fascinating as me.

![photo_2019-10-02_16-38-41](https://user-images.githubusercontent.com/43445551/66054858-ea17d000-e534-11e9-896b-6da09756beda.jpg)

![photo_2019-10-02_16-38-44](https://user-images.githubusercontent.com/43445551/66054900-f865ec00-e534-11e9-8314-3aff53e19969.jpg)

![Support tests Z0 0 4 Nozzle](https://user-images.githubusercontent.com/43445551/66054926-061b7180-e535-11e9-8e47-9757f3e4e774.JPG)

**Comment by smartavionics** [comment]
If you want walls to be treated specially as well you may as well just use the bridge settings. Currently, the presence of support stops the bridge settings being used but I could add a setting that makes it ignore the support so the skins and the walls above the support would be treated as bridges. In that case, the settings I added today are no longer useful and I will remove them again.

**Comment by kulfuerst** [comment]
I would be very happy to test that out :)

**Comment by smartavionics** [comment]
OK, I've removed the settings I added earlier and introduced a new Bridge Over Support setting that ignores the support as far as bridging is concerned. I just updated the same 1002 file so download it again.

**Comment by kulfuerst** [comment]
I think I have spotted a bug in the bridge skin generation and bridge wall generation. As you can see in the images, Cura generates bridge skin in an area where it is actually supported and shouldn't be generated (image 1). It also, the next two layers up, generates bridge walls where they are actually supported (image 2 & 3).
If I select Bridge Has Multiple Layers, Cura tries to cram all the bridge layers into one (image 4).
I'm attaching the STL, gcode and project file below.

![image](https://user-images.githubusercontent.com/43445551/66127178-67058100-e5eb-11e9-927b-bd33e2f0fc99.png)

![image](https://user-images.githubusercontent.com/43445551/66127222-7a185100-e5eb-11e9-9483-b05fb0c30be8.png)

![image](https://user-images.githubusercontent.com/43445551/66127253-869ca980-e5eb-11e9-9b31-4053eeeaf8db.png)

![image](https://user-images.githubusercontent.com/43445551/66127377-c6639100-e5eb-11e9-946d-2b0f8dc54be9.png)

[bridge generation bug.zip](https://github.com/smartavionics/Cura/files/3686063/bridge.generation.bug.zip)

**Comment by smartavionics** [comment]
I agree that something weird is happening when the bridge has multiple layers as it appears to overlap skin regions. I will look into that. However, I think the behaviour for a single bridge layer is OK. There really isn't any support under the region where it bridges so the bridge is entirely reasonable, isn't it?

**Comment by kulfuerst** [comment]
I agree that from a practical standpoint it may seems like useful behavior in some cases. In my arguably very special use case right now this is not welcome however as I need the top layers to bond very well to the bridge walls. I believe that is behavior that is also generally desirable so the bridge walls become an integral part of the whole print with uniform strength.
Besides the practicality of it, it appears to be in conflict with how bridge walls are defined, as unsupported, and it appears arbitrary too. In plenty of other regions of the printed object, walls placed on top of bridge walls are printed with normal wall settings. The fact that the behavior is unpredictable is making it difficult to use.

**Comment by kulfuerst** [comment]
Ohhh there may be a misunderstanding here. I'm talking about Cura putting bridge wall on top of bridge wall. The lower bridge walls are of course absolutely justified. I've got a better image of it here:

![image](https://user-images.githubusercontent.com/43445551/66135136-bfdc1600-e5f9-11e9-8886-dc513e057239.png)

**Comment by smartavionics** [comment]
It doesn't for me, as you can see in this image, the wall that is immediately above the outer bridge wall on the layer below is not itself printed as a bridge wall, it's normal wall. The walls that are outside of that wall are bridged but that's because they are above air.

![Screenshot_2019-10-03_15-57-56](https://user-images.githubusercontent.com/585618/66138603-cf0d9480-e5f6-11e9-8af2-15e676588221.png)

**Comment by smartavionics** [comment]
Ah, perhaps you are talking about the skin lines on the layer above as shown here. I will look into that...

![Screenshot_2019-10-03_16-02-20](https://user-images.githubusercontent.com/585618/66138869-48a58280-e5f7-11e9-9c3b-82b16778fae0.png)

**Comment by kulfuerst** [comment]
That is exactly what I meant! I really should've gone into Line Type view. I thought those were bridge walls, not bridge skin. Unwanted bridge skin also shows up in earlier layers like 51 - 54:

![image](https://user-images.githubusercontent.com/43445551/66139195-4562c480-e600-11e9-8671-28c8f51fe029.png)

**Comment by smartavionics** [comment]
I think I have fixed this, please try 1003 release and let me know if there are further issues. Thanks.

**Comment by kulfuerst** [comment]
That did fix both issues indeed! I'm very impressed.
Btw, in case it is of interest. When Bridge Over Supports in enabled it is possible to print with a Support Z Distance of 0 when using layer heights of 0.2 mm or greater. Important is very low flow (50% or less), density of about 75% and 100% fan speed. Then the supports are easy to remove and the surface finish is brilliant :)
Still working on getting it to work with lower layer height.

**Comment by smartavionics** [comment]
Thanks for the info, sounds very promising.
